libs/application-generic/src/usecases/get-preferences/get-preferences.usecase.ts

```diff
@@ -171,6 +173,29 @@ export class GetPreferences {
       return { preferences: undefined, type: undefined, source };
     }
 
+    const readOnlyFlag = workflowPreferences?.all?.readOnly;
+
+    // Determine the most specific preference applied
+    let mostSpecificPreference: PreferencesTypeEnum | undefined;
+    if (subscriberWorkflowPreferences && !readOnlyFlag) {
+      mostSpecificPreference = PreferencesTypeEnum.SUBSCRIBER_WORKFLOW;
+    } else if (subscriberGlobalPreferences && !readOnlyFlag) {
```

Comment:
Additional check of the `readOnlyFlag` at the Workflow level to conditionally scope the type, used in `send-message` to determine which execution detail to record.